server/src/main/java/org/elasticsearch/rest/action/admin/indices/RestCreateIndexAction.java

```diff
@@ -77,9 +65,8 @@ public RestChannelConsumer prepareRequest(final RestRequest request, final NodeC
     }
 
 
-    static Map<String, Object> prepareMappings(Map<String, Object> source, boolean includeTypeName) {
-        if (includeTypeName
-            || source.containsKey("mappings") == false
+    static Map<String, Object> prepareMappings(Map<String, Object> source) {
+        if (source.containsKey("mappings") == false
```

Comment:
here also on the line 80, may be also worth correcting  the error message to have only "The mapping definition cannot be nested under a type"